builder/dockerfile/parser/parser.go

```diff
@@ -204,6 +213,13 @@ func Parse(rwc io.Reader, d *Directive) (*Node, error) {
 		}
 
 		if child != nil {
+			if warning {
+				fmt.Fprintf(stderr, "[WARNING]: Empty lines detected in the following instruction:\n")
+				for _, line := range strings.Split(scannedLine, "\n") {
+					fmt.Fprintf(stderr, "[WARNING]:   %s\n", line)
+				}
+				fmt.Fprintf(stderr, "[WARNING]: Empty lines inside a RUN instruction is deprecated, and will be removed in Docker 1.16.\n")
```

Comment:
This would probably be better off in a new function, it's not directly related to parsing.

```go
if blankLineWithinInstruction {
    warnOnBlankLinWithinInstruction(stderr, scannedLines)
}
```